lib/arrowController.js

```diff
@@ -0,0 +1,190 @@
+    let arrowPath = arrow1x;
+
+    switch(this._options.scale) {
+      case 1.5:
+        arrowPath = arrow15x;
+        break;
+      case 2.0:
+        arrowPath = arrow2x;
+        break;
+      case 3.0:
+        arrowPath = arrow3x;
+        break;
+      case 4.0:
+        arrowPath = arrow4x;
+        break;
+      default:
+        arrowPath = arrow1x;
+        break;
+    }
+
+    this._arrow = new fabric.Path(arrowPath, {
+      originX: 'center',
+      originY: 'center',
+      hasBorders: false,
+      hasControls: false,
+      lockScalingX: true,
+      lockScalingY: true,
+      lockRotation: true,
+      angle: -45,
+      width: 20 * this._options.scale,
+      height: 20 * this._options.scale,
+      fill: this._options.fillColor,
+      className: this,
+    });
+
+    this._line.delete = this._arrow.delete = () => this.delete();
+
+    this.setFillColor = this.setFillColor.bind(this);
+    this.delete = this.delete.bind(this);
+    this._calcArrowAngle = this._calcArrowAngle.bind(this);
+    this._moveEnd = this._moveEnd.bind(this);
+    this._moveLine = this._moveLine.bind(this);
+    this._onMouseDown = this._onMouseDown.bind(this);
+    this._onMouseMove = this._onMouseMove.bind(this);
+    this._onMouseUp = this._onMouseUp.bind(this);
+  }
+
+  setFillColor(color) {
+    this._options.fillColor = color;
+  }
+
+  delete() {
+    this._line.remove();
+    this._arrow.remove();
+  }
+
+  _calcArrowAngle(x1, y1, x2, y2) {
+    let angle = 0;
+
+    const x = (x2 - x1);
+    const y = (y2 - y1);
+
+    if (x === 0) {
+      angle = (y === 0) ? 0 : (y > 0) ? Math.PI / 2 : Math.PI * 3 / 2;
+    } else if (y === 0) {
+      angle = (x > 0) ? 0 : Math.PI;
+    } else {
+      angle = (x < 0) ? Math.atan(y / x) + Math.PI : (y < 0) ? Math.atan(y / x) + (2 * Math.PI) : Math.atan(y / x);
+    }
+
+    return (angle * 180 / Math.PI) - 90;
+  }
+
+  _moveEnd() {
+    this._line.set({
+      'x2': this._arrow.left,
+      'y2': this._arrow.top,
+    });
+    this._line._setWidthHeight();
+    this._line.setCoords();
+
+    this._arrow.set('angle', this._calcArrowAngle(this._line.x2, this._line.y2, this._line.x1, this._line.y1));
+  }
+
+  _moveLine() {
+    const oldCenterX = (this._line.x1 + this._line.x2) / 2;
+    const oldCenterY = (this._line.y1 + this._line.y2) / 2;
+    const deltaX = this._line.left - oldCenterX;
+    const deltaY = this._line.top - oldCenterY;
+
+    this._arrow.set({
+      'left': this._line.x2 + deltaX,
+      'top': this._line.y2 + deltaY,
+    }).setCoords();
+
+    this._line.set({
+      'x1': this._line.x1 + deltaX,
+      'y1': this._line.y1 + deltaY,
+      'x2': this._line.x2 + deltaX,
+      'y2': this._line.y2 + deltaY,
+    });
+
+    this._line.set({
+      'left': (this._line.x1 + this._line.x2) / 2,
+      'top': (this._line.y1 + this._line.y2) / 2,
+    });
+  }
+
+  _onMouseDown(that, o) {
```

Comment:
A bit of nailpolish here but can we use more descriptive variable names instead of `that`? If this is an element we should call it `element` or `wrapper`, etc.